### PR TITLE
Drop Python 3.7 and add Address fields

### PR DESCRIPTION
**Pull request by risktoparkrageram**
Some small fixes from using pyap;
1. The `Address` class has fields that mypy wants to know exist.
2. some of the dependencies are causing issues downstream for us, so I'm dropping python 3.7 support and loosening the requirements to have more up to date versions.

### CODE REVIEW

File: pyap_beauhurst/address.py [hunk 1]
```diff
@@ -12,10 +12,23 @@
 
 
 class Address:
-    full_address: str
-    state: str
+    building_id: str
```
Comment (janrito) [inline pyap_beauhurst/address.py:15]: Seems like this should be replaced with a dataclass and a validation function (or even pydantic)

File: pyproject.toml [hunk 2]
```diff
@@ -22,15 +22,15 @@ readme="README.rst"
 python = "^3.7.1"
 
 [tool.poetry.dev-dependencies]
-black = ">=21.5b1,<24"
-ipython = ">=7.0,<9.0"
-mypy = ">=0.901,<1"
-pytest = ">=6.2.4,<8"
-pytest-cov = ">=2.12.1,<4.1"
-pytest-mock = ">=3.6.1,<3.11"
+black = "^21.5b1"
```
Comment (janrito) [inline pyproject.toml]: doesn't this top it at v22?

File: .github/workflows/release.yaml [hunk 3]
```diff
@@ -14,7 +14,7 @@ jobs:
       - name: Set up Python
         uses: actions/setup-python@v4
         with:
-          python-version: "3.7"
```
Comment (janrito) [inline .github/workflows/release.yaml:17]: will we not need python 3.7 for mwaa?

Comment (risktoparkrageram) [reply .github/workflows/release.yaml:17]: We are not using the lib on mwaa, only on the lambda now.

File: pyproject.toml (hunk 2 above)
Comment (risktoparkrageram) [reply pyproject.toml]: had a bit of a fight, but we are now I think in a better place

File: pyap_beauhurst/address.py [hunk 4]
```diff
@@ -8,33 +8,43 @@
...
+    def strip_chars(cls, v: Any) -> Any:
+        if isinstance(v, str):
+            return v.strip(" ,;:")
+        if v:
+            return v
 
     def as_dict(self) -> Dict[str, str]:
         # Return parsed address parts as a dictionary
-        return self.data_as_dict
+        return self.dict()
```
[the remark this answers is not in the document]
Comment (risktoparkrageram) [reply pyap_beauhurst/address.py]: no not really, I was just keeping the API the same where possible.

[the remark this answers is not in the document]
Comment (risktoparkrageram) [reply pyap_beauhurst/address.py]: break things: 95ef84964a490b2e5e922c6da18714f67cc33909 :hammer:

**Review by janrito — APPROVED** [review]

**Comment by risktoparkrageram** [comment]
thanks @janrito

**Pull request merged**